### PR TITLE
fix(ui): hide elasticSearchIndex page for non-admins

### LINKED ISSUE #7914: [ElasticSearchReindex] Non-Admin users should not see the Elastic Search Reindex tab
**Affected module**
Does it impact the UI, backend or Ingestion Framework?
UI

**Describe the bug**
Currently the non-admin users can see the elastic search tab in settings with reindexing button disabled, we should not allow that

**To Reproduce**

Screenshots or steps to reproduce

**Expected behavior**
Non-Admin users should not see the Elastic Search Reindex tab

**Version:**
 - Python version:
 - OpenMetadata version: [e.g. 0.8]
 - OpenMetadata Ingestion package version: [e.g. `openmetadata-ingestion[docker]==XYZ`]

**Additional context**

### PR DESCRIPTION
**Pull request by chirag-madlani**
### Describe your changes :

Closes #7914 

#
### Type of change :

- [x] Bug fix
- [x] Improvement
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Frontend Preview (Screenshots) :
Settings page for Admin
Screenshot 2022-11-30 at 5 46 51 PM

Settings page for Non Admin
Screenshot 2022-11-30 at 5 46 35 PM

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/utils/GlobalSettingsUtils.tsx [hunk 1]
```diff
@@ -288,7 +292,11 @@ export const getGlobalSettingMenuItem = (
 } => {
   const subItems = children
     ? children
-        .filter((menu) => menu.isProtected)
+        .filter((menu) =>
+          menu.isAdmin
+            ? menu.isProtected
+            : menu.isProtected && isUndefined(menu.isAdmin)
+        )
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/utils/GlobalSettingsUtils.tsx:299]:
```suggestion
          menu.isAdmin
            ? menu.isAdmin
            : menu.isProtected 
        )
```

File: openmetadata-ui/src/main/resources/ui/src/utils/GlobalSettingsUtils.tsx [hunk 2]
```diff
@@ -225,6 +227,7 @@ export const getGlobalSettingsMenuWithPermission = (
           icon: (
             <ElasticSearchIcon className="tw-w-4 tw-mt-1.5 side-panel-icons" />
           ),
+          isAdmin: isAdminUser,
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/utils/GlobalSettingsUtils.tsx:230]: We have `isProtected` flag so can we use that?

File: openmetadata-ui/src/main/resources/ui/src/utils/GlobalSettingsUtils.tsx (hunk 1 above)
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/utils/GlobalSettingsUtils.tsx:299]: We don't need this logic if we just use the `isProtected` flag.

**Review by Sachin-chaurasiya — CHANGES_REQUESTED** [review]

File: openmetadata-ui/src/main/resources/ui/src/utils/GlobalSettingsUtils.tsx (hunk 2 above)
Comment (chirag-madlani) [reply openmetadata-ui/src/main/resources/ui/src/utils/GlobalSettingsUtils.tsx]: Done

File: openmetadata-ui/src/main/resources/ui/src/utils/GlobalSettingsUtils.tsx (hunk 1 above)
Comment (chirag-madlani) [reply openmetadata-ui/src/main/resources/ui/src/utils/GlobalSettingsUtils.tsx]: Done

**Review by ShaileshParmar11 — DISMISSED** [review]

File: openmetadata-ui/src/main/resources/ui/src/utils/GlobalSettingsUtils.tsx [hunk 3]
```diff
@@ -277,6 +275,7 @@ export const getGlobalSettingMenuItem = (
     label: string;
     isProtected: boolean;
     icon: React.ReactNode;
+    isAdmin?: boolean;
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/utils/GlobalSettingsUtils.tsx:278]: Remove this

**Review by Sachin-chaurasiya — APPROVED** [review]

**Review by ShaileshParmar11 — APPROVED** [review]

**Pull request merged**